server/aesdsocket.c

```diff
@@ -148,7 +153,20 @@ void* receive_send_data(void* thread_param)
 	int received_bytes, write_bytes, read_bytes, send_bytes;
 	int loc = 0;
 	int count = 1;
+	int index = 0, newline_index=0;
+    	int write_size = 0;
+    	char ch;
+    	int newtransmit_buffer_size = BUFFER_SIZE;
 
+#ifdef USE_AESD_CHAR_DEVICE
```

Comment:
@midhun9 why does this code block need to be specific to `USE_AESD_CHAR_DEVICE`?  A comment about this would be good.